src/blockdata/block.rs

```diff
@@ -141,9 +141,11 @@ impl BlockHeader {
             return Err(BlockBadTarget);
         }
         let block_hash = self.block_hash();
-        let mut ret = [0u64; 4];
-        util::endian::bytes_to_u64_slice_le(block_hash.as_inner(), &mut ret);
-        let hash = &Uint256(ret);
+
+        let mut bytes = block_hash.into_inner();
+        bytes.reverse();
+
+        let hash = &Uint256::from_be_bytes(bytes);
         if hash <= target { Ok(block_hash) } else { Err(BlockBadProofOfWork) }
```

Comment:
Would prefer this to go after #1197 to decrease the noise.

Reply:
No worries, I probably wasn't careful enough doing this PR. I'll wait till #1197 gets done then re-visit this.